content/en/tracing/trace_collection/automatic_instrumentation/dd_libraries/cpp.md

```diff
@@ -23,153 +23,169 @@ further_reading:
-**Note**: C++ does not provide integrations for OOTB instrumentation, but it's used by Proxy tracing such as [Envoy][1] and [Nginx][2]. For compatibility requirements for the C++ Tracer, visit the [Compatibility Requirements][3] page.
+
+<div class="alert alert-warning">
+  <strong>Note:</strong> C++ does not provide integrations for automatic instrumentation, but it's used by Proxy tracing such as <a href="/tracing/setup/envoy/">Envoy</a> and <a href="/tracing/setup/nginx/">Nginx</a>.
+</div>
+
+## Compatibility requirements
+The C++ tracing library requires C++17 toolchain to build. For a full list of Datadog's tracing library compatiblity requirements and processor architecture support, visit the [Compatibility Requirements][3] page.
+
+## Getting started
+Before you begin, make sure you have already [installed and configured the Agent][6].
 
 ## Instrument your application
 
-After [the Agent is installed][6], follow these steps to add the Datadog tracing library to your C++ applications in one of two ways:
+{{< tabs >}}
 
-* Compile against dd-opentracing-cpp, where the Datadog lib is compiled in and configured in code
-* Dynamic loading, where the Datadog OpenTracing library is loaded at runtime and configured via JSON
+{{% tab "CMake" %}}
+````CMake
+include(FetchContent)
 
-### Compile against dd-opentracing-cpp
+FetchContent_Declare(
+  dd-trace-cpp
+  GIT_REPOSITORY https://github.com/DataDog/dd-trace-cpp
+  GIT_TAG        v0.2.0
+  GIT_SHALLOW    ON
+  GIT_PROGRESS   ON
+)
 
-```bash
-# Requires the "jq" command, which can be installed via
-# the package manager, for example "apt install jq",
-# "apk add jq", "yum install jq".
-if ! command -v jq >/dev/null 2>&1; then
-  >&2 echo "jq command not found. Install using the local package manager."
-else
-  # Gets the latest release version number from GitHub.
-  get_latest_release() {
-    curl --silent "https://api.github.com/repos/$1/releases/latest" | jq --raw-output .tag_name
-  }
-  DD_OPENTRACING_CPP_VERSION="$(get_latest_release DataDog/dd-opentracing-cpp)"
-  # Download and install dd-opentracing-cpp library.
-  wget https://github.com/DataDog/dd-opentracing-cpp/archive/${DD_OPENTRACING_CPP_VERSION}.tar.gz -O dd-opentracing-cpp.tar.gz
-  mkdir -p dd-opentracing-cpp/.build
-  tar zxvf dd-opentracing-cpp.tar.gz -C ./dd-opentracing-cpp/ --strip-components=1
-  cd dd-opentracing-cpp/.build
-  # Download and install the correct version of opentracing-cpp, & other deps.
-  ../scripts/install_dependencies.sh
-  # Configure the project, build it, and install it.
-  cmake ..
-  make -j
-  make install
-fi
-```
+FetchContent_MakeAvailable(dd-trace-cpp)
+
+# Add `tracer_example` target
+add_executable(tracer_example tracer_example.cpp)
 
-Include `<datadog/opentracing.h>` and create the tracer:
+# Statically link against `dd-trace-cpp`
+# NOTE: To dynamically link against `dd-trace-cpp` use the `dd_trace_cpp_shared` target
+target_link_libraries(tracer_example dd_trace_cpp-static)
+````
 
```

Comment:
```suggestion
Here's an example of how to use the dd-trace-cpp library in your C++ code to create spans and traces:
```
^ I think we need to add more introduction to these code examples so it's clear for users. Something like this (please check).